pokemon.js

```diff
@@ -0,0 +1,111 @@
+const pokeAPI = 'https://pokeapi.co/api/v2/pokemon';
+const pokemonsData = fetch (pokeAPI);
+const pokemonCards = document.querySelector('[data-pokemon-cards]');
+const pokemonWebsite = document.querySelector('[data-pokemon-website]');
+const progressItem = document.createElement('progress');
+
+//function to create cards with pokemon names
+
+function getPokeNames() {
+pokemonsData.then( response => {
+    if(!response.ok) {
+        pokemonCards.innerText = 'Failed to load data...';
+        progressItem.classList.add('disabled');
+    }
+progressItem.setAttribute('id', 'pokemons');
+pokemonWebsite.prepend(progressItem);
+pokemonCards.innerText = 'Loading...';
+return response.json();
+}).then (pokemons => {
+    progressItem.classList.add('disabled');
+    pokemonCards.innerText = '';
+    let pokemonResults = pokemons.results;
+    for(let i = 0; i < pokemonResults.length; i++) {
+        pokemonsData.then(
+        pokemon => {
+            let pokemonName = pokemonResults[i].name;
+            const pokemonDOMElement = document.createElement('div');
+            pokemonDOMElement.setAttribute('id', i+1);
+            pokemonDOMElement.innerText = pokemonName;
+            pokemonDOMElement.classList.add('one-pokemon');
+            pokemonCards.appendChild(pokemonDOMElement); 
+        }
+        ).catch((error) => {
+            console.log('promise error', error);
+            pokemonCards.innerText = 'Failed to load data...';
+        });
+    }
+    return;
+})};
+
+//function to add photo and experience to every pokemon
+
+function pokemonsDetails(id) {
+    let onePoke = fetch (pokeAPI + '/' + id);
+    onePoke.then (
+        response => {
+            return response.json();
+        }).then(onePoke => {
+            pokemonDOMElement = document.getElementById(id);
+            let pokemonPower = document.createElement('div');
+            pokemonPower.setAttribute('id', id + 'power');
```

Comment:
tutaj w tym then podobnie jak wyżej, wydzieliłbym to do funkcji która by przyjmowała odpowiednie parametry a w tym miejscu ją bym wywoływał